tests/foreman/cli/test_provisioning.py

```diff
@@ -15,15 +15,428 @@
+    repo_id = enable_rhrepo_and_fetchid(
+            basearch=REPOS['rhel8ksappstream']['arch'],
+            org_id=org.id,
+            product=REPOS['rhel8ksappstream']['product'],
+            releasever=REPOS['rhel8ksappstream']['releasever'],
+            repo=REPOS['rhel8ksappstream']['name'],
+            reposet=REPOS['rhel8ksappstream']['reposet'],
+    )
+    repo = entities.Repository(user_credentials, id=repo_id)
+    response = repo.sync(synchronous=False)
+    repo1_sync_task = response['id']
+    # Now RHEL8 BaseOS KS which we will return at the end
+    repo_id = enable_rhrepo_and_fetchid(
+            basearch=REPOS['rhel8ksbaseos']['arch'],
+            org_id=org.id,
+            product=REPOS['rhel8ksbaseos']['product'],
+            releasever=REPOS['rhel8ksbaseos']['releasever'],
+            repo=REPOS['rhel8ksbaseos']['name'],
+            reposet=REPOS['rhel8ksbaseos']['reposet'],
+    )
+    repo = entities.Repository(user_credentials, id=repo_id)
+    response = repo.sync(synchronous=False)
+    repo2_sync_task = response['id']
+    # Wait for sync to finish
+    entities.ForemanTask(id=repo1_sync_task).poll(timeout=1800)
+    entities.ForemanTask(id=repo2_sync_task).poll(timeout=100)
+    return repo
+
+
+@pytest.fixture(scope='session')
+def cv(user_credentials, org, rhel7, rhel7ks):
+    # TODO: Some puppet modules in here would be nice
+    content_view = entities.ContentView(
+        user_credentials,
+        organization=org
+    ).create()
+
+    content_view.repository = [rhel7, rhel7ks]
+    content_view = content_view.update(['repository'])
+    return content_view
+
+
+@pytest.fixture(scope='session')
+def cvv(user_credentials, org, cv, lce):
+    call_entity_method_with_timeout(cv.publish, timeout=1500)
+    cv = cv.read()
+    assert len(cv.version) == 1, "Single CV version expected"
+    cv_version = cv.version[0].read()
+    assert len(cv_version.environment) == 1, "Single LCE expected for the given CV version"
+    promote(cv_version, lce.id)
+    cv = cv.read()
+    assert len(cv.version) == 1, "Single CV version expected"
+    return cv_version.read()
+
+
+@pytest.fixture(scope='session')
+def ak(user_credentials, org, cv, lce):
+    activation_key_name = gen_string('alpha')
+    activation_key = entities.ActivationKey(
+        user_credentials,
+        name=activation_key_name,
+        environment=lce,
+        organization=org,
+        content_view=cv,
+    ).create()
+    for sub in entities.Subscription(organization=org).search():
+        if sub.read_json()['name'] == DEFAULT_SUBSCRIPTION_NAME:
+            activation_key.add_subscriptions(data={
+                'quantity': 1,
+                'subscription_id': sub.id,
+            })
+            break
+    return activation_key
+
+
+@pytest.fixture(scope='session')
+def cr(user_credentials, org, loc):
+    compute_resource = entities.LibvirtComputeResource(
+        user_credentials,
+        url='qemu+ssh://root@{0}/system'.format(
```

Comment:
optional: you can remove the format indices, as we no longer have to support py2.